### PR TITLE
Remote function translate

### PR DESCRIPTION
**Pull request by shen-chao**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [*] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [*] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [*] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [*] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [*] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

File: bigquery/remote-function/translate/noxfile_config.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TEST_CONFIG_OVERRIDE copied from the source of truth:
+# https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/noxfile_config.py
+
+TEST_CONFIG_OVERRIDE = {
+    # You can opt out from the test for specific Python versions.
+    "ignored_versions": ["2.7"],
```
Comment (autoerr) [inline bigquery/remote-function/translate/noxfile_config.py:20]: Consider sync'ing with https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/noxfile_config.py. 3.6 is now excluded.

File: bigquery/remote-function/translate/main_test.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from unittest import mock
+
+
+import flask
+from google.cloud.translate_v3 import types
```
Comment (autoerr) [inline bigquery/remote-function/translate/main_test.py:18]: Consider importing translate, and then use translate.TranslateTextResponse:
https://sourcegraph.com/github.com/googleapis/python-translate/-/blob/google/cloud/translate/__init__.py?subtree=true

I used vision_v1 initially, but then customers told us it is broken for recursive import, after https://github.com/googleapis/python-vision/pull/498

Traceback (most recent call last):
  File ".../vision_function.py", line 18, in <module>
    from google.cloud import vision_v1
  File ".../Library/Python/3.9/lib/python/site-packages/google/cloud/vision_v1/__init__.py", line 16, in <module>
    from google.cloud.vision import gapic_version as package_version
  File ".../Library/Python/3.9/lib/python/site-packages/google/cloud/vision/__init__.py", line 26, in <module>
    from google.cloud.vision_v1 import ImageAnnotatorClient
ImportError: cannot import name 'ImageAnnotatorClient' from partially initialized module 'google.cloud.vision_v1' (most likely due to a circular import) (.../Library/Python/3.9/lib/python/site-packages/google/cloud/vision_v1/__init__.py)

See context at: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/8883

**Review by autoerr — APPROVED** [review]

File: bigquery/remote-function/translate/main_test.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+
+# Create a fake "app" for generating test request contexts.
+@pytest.fixture(scope='module')
+def app() -> flask.Flask:
+    return flask.Flask(__name__)
+
+
+@mock.patch('main.translate_client')
+def test_main(mock_translate: object, app: flask.Flask) -> None:
```
Comment (chalmerlowe) [inline bigquery/remote-function/translate/main_test.py:29]: Comment: we have a single test here that basically ends up calling `handle_translation()`.
`handle_translation()` calls our other functions so they are in effect being tested.
Which is potentially fine.

We don't have any tests of user supplied parameters to confirm that our function can handle reasonable edge cases. The one that comes to mind is the one for the path created by `extract_project_from_caller()`.

Again, gonna leave it up to you to determine whether additional effort is reasonable to increase the robustness of this test suite.

File: bigquery/remote-function/translate/main.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+translate_client = translate.TranslationServiceClient()
+
+
+# Register an HTTP function with the Functions Framework
+@functions_framework.http
+def handle_translation(request: flask.Request) -> flask.Response:
+    """BigQuery remote function to translate input text.
+
+    Args:
+        request: HTTP request from BigQuery
```
[the remark this answers is not in the document]
Comment (shen-chao) [reply bigquery/remote-function/translate/main.py:33]: The function here (and those in this remote_function folder) is expected to serve requests from BigQuery
https://cloud.google.com/bigquery/docs/reference/standard-sql/remote-functions

File: bigquery/remote-function/translate/main.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    Args:
+        job: full resource name of a BigQuery job, like
+          "//bigquery.googleapi.com/projects/<project>/jobs/<job_id>"
+
+    Returns:
+        project id which is contained in the full resource name of the job.
+    """
+    path = job.split("/")
+    return path[4] if len(path) > 4 else None
```
[the remark this answers is not in the document]
Comment (shen-chao) [reply bigquery/remote-function/translate/main.py:85]: Given the inputs are from BigQuery instead of users directly, I feel like it's probably fine.

File: bigquery/remote-function/translate/main_test.py (hunk 3 above)
Comment (shen-chao) [reply bigquery/remote-function/translate/main_test.py:29]: Given that the requests are from BigQuery, edge cases are probably not high priority if it's ok skip.

**Review by chalmerlowe — APPROVED** [review]
LGTM
